File: code_snippets/api-guide/workflows/common_workflows/py/workflow_predict_audio.py
###########################################################################################
# In this section, we set the user authentication, user and app ID, workflow ID, and
# audio URL. Change these strings to run your own example.
##########################################################################################

# Your PAT (Personal Access Token) can be found in the portal under Authentification
PAT = "YOUR_PAT_HERE"
USER_ID = "clarifai"
APP_ID = "main"
# Change these to make your own predictions
WORKFLOW_ID = "asr-sentiment"
AUDIO_URL = "https://samples.clarifai.com/negative_sentence_1.wav"
# Or, to use a local audio file, assign the location variable
# AUDIO_FILE_LOCATION = "YOUR_AUDIO_FILE_LOCATION_HERE"

##########################################################################
# YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
##########################################################################

from clarifai_grpc.channel.clarifai_channel import ClarifaiChannel
from clarifai_grpc.grpc.api import resources_pb2, service_pb2, service_pb2_grpc
from clarifai_grpc.grpc.api.status import status_code_pb2

channel = ClarifaiChannel.get_grpc_channel()
stub = service_pb2_grpc.V2Stub(channel)

metadata = (("authorization", "Key " + PAT),)

userDataObject = resources_pb2.UserAppIDSet(
    user_id=USER_ID, app_id=APP_ID
)  # The userDataObject is required when using a PAT

# To use a local video file, uncomment the following lines
# with open(AUDIO_FILE_LOCATION, "rb") as f:
# audio_bytes = f.read()

post_workflow_results_response = stub.PostWorkflowResults(
    service_pb2.PostWorkflowResultsRequest(
        user_app_id=userDataObject,
        workflow_id=WORKFLOW_ID,
        inputs=[
            resources_pb2.Input(
                data=resources_pb2.Data(
                    audio=resources_pb2.Audio(
                        url=AUDIO_URL,
                        # base64=audio_bytes
                    )
                )
            )
        ],
    ),
    metadata=metadata,
)
if post_workflow_results_response.status.code != status_code_pb2.SUCCESS:
    print(post_workflow_results_response.status)
    raise Exception(
        "Post workflow results failed, status: "
        + post_workflow_results_response.status.description
    )

# We'll get one WorkflowResult for each input we used above. Because of one input, we have here one WorkflowResult
results = post_workflow_results_response.results[0]

# Each model we have in the workflow will produce its output
for output in results.outputs:
    model = output.model
    print("Output for the model: `%s`" % model.id)
    for concept in output.data.concepts:
        print("\t%s %.2f" % (concept.name, concept.value))
    print(output.data.text.raw)

# Uncomment this line to print the full Response JSON
# print(results)

File: code_snippets/api-guide/workflows/common_workflows/node/workflow_predict_audio.js

//index.js file

//////////////////////////////////////////////////////////////////////////////////////////
// In this section, we set the user authentication, user and app ID, workflow ID, and
// audio URL. Change these strings to run your own example.
/////////////////////////////////////////////////////////////////////////////////////////

// Your PAT (Personal Access Token) can be found in the Account's Security section
const PAT = "YOUR_PAT_HERE";
const USER_ID = "clarifai";
const APP_ID = "main";
// Change these to make your own predictions
const WORKFLOW_ID = "asr-sentiment";
const AUDIO_URL = "https://samples.clarifai.com/negative_sentence_1.wav";
// Or, to use a local audio file, assign the location variable
// const AUDIO_FILE_LOCATION = "YOUR_AUDIO_FILE_LOCATION_HERE";

/////////////////////////////////////////////////////////////////////////////
// YOU DO NOT NEED TO CHANGE ANYTHING BELOW THIS LINE TO RUN THIS EXAMPLE
/////////////////////////////////////////////////////////////////////////////

const { ClarifaiStub, grpc } = require("clarifai-nodejs-grpc");

const stub = ClarifaiStub.grpc();

// This will be used by every Clarifai endpoint call
const metadata = new grpc.Metadata();
metadata.set("authorization", "Key " + PAT);

// To use a local text file, uncomment the following lines
// const fs = require("fs");
// const audioBytes = fs.readFileSync(AUDIO_FILE_LOCATION);

stub.PostWorkflowResults(
    {
        user_app_id: {
            "user_id": USER_ID,
            "app_id": APP_ID,
        },
        workflow_id: WORKFLOW_ID,
        inputs: [{
            data: {
                audio: {
                    url: AUDIO_URL,
                    // base64: audioBytes
                }
            }
        }],
    },
    metadata,
    (err, response) => {
        if (err) {
            throw new Error(err);
        }

        if (response.status.code !== 10000) {
            throw new Error(
                "Post workflow results failed, status: " + response.status.description
            );
        }

        // We'll get one WorkflowResult for each input we used above. Because of one input, we have here 
        // one WorkflowResult
        const results = response.results[0];

        // Each model we have in the workflow will produce its output   
        for (const output of results.outputs) {
            const model = output.model;
            console.log("Output for the model: `" + model.id + "`");
            for (const concept of output.data.concepts) {
                console.log("\t" + concept.name + " " + concept.value);
            }
            if (output.data.text) {
                console.log(output.data.text.raw);
            }
        }
    }
);
